pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -343,39 +399,308 @@ func (r *ReconcileVirtualMachineImport) addWatchForImportPod(instance *v2vv1.Vir
+		if err != nil {
+			return false, err
+		}
+	}
+
+	job, err := r.findGuestConversionJob(vmName)
+	if err != nil {
+		return false, err
+	}
+	// the job doesn't exist, so create it
+	if job == nil {
+		job = r.makeGuestConversionJobSpec(vmSpec)
+		// Set VirtualMachineImport instance as the owner and controller so it'll be cleaned up
+		// when the instance is removed.
+		if err := controllerutil.SetControllerReference(instance, job, r.scheme); err != nil {
+			return false, err
+		}
+		err := r.client.Create(context.TODO(), job)
+		if err != nil {
+			return false, err
+		}
+
+		processingCond := conditions.NewProcessingCondition(string(v2vv1.ConvertingGuest), "Running virt-v2v", corev1.ConditionTrue)
+		err = r.upsertStatusConditions(types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, processingCond)
+		if err != nil {
+			return false, err
+		}
+
+		// Update progress to converting guest:
+		if err = r.updateProgress(instance, progressConvertingGuest); err != nil {
+			return false, err
+		}
+	}
+
+	if job.Status.Active > 0 {
+		return false, nil
+	}
+
+	if job.Status.Failed > 0 {
+		err := r.endGuestConversionAsFailed(provider, instance, "virt-v2v job failed")
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}
+
+	return job.Status.Succeeded > 0, nil
+}
+
+// findGuestConversionJob finds the guest conversion job for a given VM, returning nil if it can't be found.
+func (r *ReconcileVirtualMachineImport) findGuestConversionJob(vmName types.NamespacedName) (*batchv1.Job, error) {
+	jobList := &batchv1.JobList{}
+	matchingLabels := client.MatchingLabels(map[string]string{VMLabel: vmName.Name})
+	err := r.client.List(context.TODO(), jobList, matchingLabels)
+	if err != nil {
+		return nil, err
+	}
+	if len(jobList.Items) > 0 {
+		return &jobList.Items[0], nil
+	}
+	return nil, nil
+}
+
+func (r *ReconcileVirtualMachineImport) makeGuestConversionJobSpec(vmSpec *kubevirtv1.VirtualMachine) *batchv1.Job {
+	// Only ever run the guest conversion job once per VM
+	completions := int32(1)
+	parallelism := int32(1)
+	backoffLimit := int32(0)
+
+	volumes, volumeMounts := makeJobVolumeMounts(vmSpec)
+
+	return &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: virtV2vJobName + "-",
+			Namespace:    vmSpec.Namespace,
+			Labels: map[string]string{
+				VMLabel: vmSpec.Name,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Completions:  &completions,
+			Parallelism:  &parallelism,
+			BackoffLimit: &backoffLimit,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: virtV2vJobName + "-",
+					Namespace:    vmSpec.Namespace,
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Name:            virtV2vJobName,
+							Image:           virtV2vImage,
+							ImagePullPolicy: v1.PullIfNotPresent,
```

Comment:
`ImagePullPolicy` should be provided we should not hardcode it. There is a param that we can use here.